weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/sections/ExampleCompareSection/exampleCompareSectionUtil.ts

```diff
@@ -417,6 +417,24 @@ export function useExampleCompareData(
 
       const selectedRowDigest = targetRow.inputDigest;
 
+      // If we already have the data in cache, no need to fetch
+      if (selectedRowDigest in cachedRowData.current) {
```

Comment:
i think this early return is not quite correct as it prevents the adjacent loading (makes the assumption that if one row is preloaded, they all are). I think this entire change should just be 

```
      // Check if we can get the data from the results directly
      // This is specifically for imperative evaluations
      const preloadedDataRow = state.loadableComparisonResults.result?.inputs?.[selectedRowDigest]
      if (!(selectedRowDigest in cachedRowData.current) && preloadedDataRow) {
        const inputData = preloadedDataRow.val;
        cachedRowData.current[selectedRowDigest] = inputData;
        increaseCacheVersion();
        return;
      }
```